portal-web/test/functional/com/liferay/portalweb/macros/headless/objects/ObjectDefinitionAPI.macro

```diff
@@ -0,0 +1,316 @@
+definition {
+
+	macro _assertEmployeeFirstnameCorrect {
+		var actualEmployeeFirstname = ObjectDefinitionAPI._getObjectFirstnameById(
```

Comment:
For these macros, I'd recommend calling `Variables.assertDefined()` and defining the parameters that are required for the macro. To me it seems really confusing to know what parameters will be required from the macro/API so having `Variables.assertDefined()` called at beginning of macros will I think make it more clear to future QA Tester who may want to reuse these macros.

ie: 
```
	macro addPortalInstance {
		Variables.assertDefined(parameterList = "${domain},${virtualHost},${portalInstanceId}");
```

Reply:
Yeah, that's a good point! It is a bit painful to write but it definitely improves the readability! Thanks :)